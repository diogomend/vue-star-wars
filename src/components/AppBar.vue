<template>
  <v-app-bar color="orange" dark app>
    <v-app-bar-nav-icon @click="toggleMenu"></v-app-bar-nav-icon>
    <v-toolbar-title>Star Wars App</v-toolbar-title>
    <v-spacer></v-spacer>
  </v-app-bar>
</template>

<script>
export default {
  name: "AppBar",
  methods: {
    toggleMenu() {
      this.$emit("toggleMenu");
    }
  }
};
</script>
