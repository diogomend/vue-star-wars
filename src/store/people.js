import swapi from "@/apis/swapi";
import storageHelper from "@/helpers/localStorage";

const actions = {
  fetchPeople: async ({ commit }) => {
    const storedPeople = storageHelper.getFromStorage();

    if (storedPeople && storedPeople.length) {
      commit("SET_PEOPLE", JSON.parse(storedPeople));
      return;
    }
    await getAllPeople(commit);
  }
};

const mutations = {
  SET_PEOPLE(state, people) {
    state.people = people;
  }
};

const getters = {
  getPeople: state => state.people
};

const state = () => ({
  people: []
});

export default {
  namespaced: true,
  getters,
  actions,
  state,
  mutations
};

export const getAllPeople = async commit => {
  let currentPage = 1;
  let hasNextPage = true;
  let people = [];

  while (hasNextPage) {
    const { data } = await swapi.get("people", {
      params: {
        page: currentPage
      },
      useCache: true
    });

    people = people.concat(data.results);
    storageHelper.setItem(JSON.stringify(people));
    commit("SET_PEOPLE", people);

    hasNextPage = !!data.next;
    currentPage += 1;
  }

  return people;
};
