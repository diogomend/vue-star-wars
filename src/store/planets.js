import swapi from "@/apis/swapi";

const actions = {
  showPlanetDetails: async ({ commit }, planetID) => {
    commit("SET_SHOW_PLANET_MODAL", true);
    const { data } = await swapi.get(`planets/${planetID}`);
    commit("SET_SELECTED_PLANET", data);
  },
  hidePlanetModal: async ({ commit }) => {
    commit("SET_SHOW_PLANET_MODAL", false);
  }
};
const mutations = {
  SET_SELECTED_PLANET(state, planet) {
    state.selectedPlanet = planet;
  },
  SET_SHOW_PLANET_MODAL(state, show) {
    state.showPlanetModal = show;
  }
};

const getters = {
  getActivePlanet: state => state.selectedPlanet,
  showPlanetModal: state => state.showPlanetModal
};

const state = () => ({
  selectedPlanet: null,
  showPlanetModal: false
});

export default {
  namespaced: true,
  getters,
  actions,
  state,
  mutations
};
