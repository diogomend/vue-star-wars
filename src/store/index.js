import Vue from "vue";
import Vuex from "vuex";
import people from "@/store/people";
import planets from "@/store/planets";

Vue.use(Vuex);

export const storeOptions = {
  modules: {
    people,
    planets
  }
};

export default new Vuex.Store(storeOptions);
