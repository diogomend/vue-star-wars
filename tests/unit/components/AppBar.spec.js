import Vuetify from "vuetify";
import Vue from "vue";
import Vuex from "vuex";
import AppBar from "@/components/AppBar.vue";
import { shallowMount } from "@vue/test-utils";
import { storeOptions } from "@/store/index";
Vue.use(Vuex);
Vue.use(Vuetify);

describe("AppBar.vue", () => {
  let store;

  beforeEach(() => {
    jest.clearAllMocks();
    store = new Vuex.Store(storeOptions);
  });

  it("is a Vue instance", () => {
    const wrapper = shallowMount(AppBar, {
      store
    });

    const template = wrapper.html();
    expect(template).toMatchSnapshot();
  });

  it("toggleMenu should emit an event", () => {
    const wrapper = shallowMount(AppBar, {
      store
    });

    wrapper.vm.toggleMenu();
    expect(Object.keys(wrapper.emitted()).length).toBe(1);
    expect(Object.keys(wrapper.emitted())[0]).toBe("toggleMenu");
  });
});
