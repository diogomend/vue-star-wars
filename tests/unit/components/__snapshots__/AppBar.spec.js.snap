// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar.vue is a Vue instance 1`] = `
<v-app-bar-stub color="orange" dark="true" tag="header" extensionheight="48" src="" value="true" app="true">
  <v-app-bar-nav-icon-stub></v-app-bar-nav-icon-stub>
  <v-toolbar-title-stub>Star Wars App</v-toolbar-title-stub>
  <v-spacer-stub></v-spacer-stub>
</v-app-bar-stub>
`;
